lib/Fhp/Model/TanRequestChallengeImage.php

```diff
@@ -25,15 +25,33 @@ public function __construct(Bin $bin)
         $mimeTypeLength = ord($mimeTypeLengthString[0]) * 256 + ord($mimeTypeLengthString[1]);
 
         $this->mimeType = substr($data, 2, $mimeTypeLength);
-        $this->data = substr($data, 2 + $mimeTypeLength + 2);
+
+        $data = substr($data, 2 + $mimeTypeLength);
+
+        $dataLengthString = substr($data, 0, 2);
+        $expectedDataLength = ord($dataLengthString[0]) * 256 + ord($dataLengthString[1]);
+        $isDataLength = strlen($data) - 2;
+
+        if($expectedDataLength != $isDataLength)
+        {
+            // This exception is thrown, if there is an encoding problem
+            // f.e.: the serialized action was saved as a string, but not base64 encoded
+            throw new \RuntimeException(
+            'Unexpected data length ' .
```

Comment:
Could this use `""` with interpolated strings?

Reply:
In my opinion, it's bad coding style to use interpolated strings. Do you insist?

Reply:
I don't insist. Why is it bad coding style? Does it have any disadvantage? It's one of the features that many non-scripting languages are adopting from the scripting langauges, so it can't be that bad.

Reply:
There are multiple reasons why it's better to use always single quotes, like 
- if working with PHP/HTML/SQL mixed codes to prevent the usage of escape characters
- if characters like "\\" should be usable, but finally not as an escape character
- if it's important, because of security reasons, to find accidentally/unwanted added variables within interpolated strings. Especially if the editor does not support the highlighting of variables within a string (like "git diff" results, Atom, Sourcetree, Windows Editor ;-) )

Reply:
> There are multiple reasons why it's better to use always single quotes, like
> 
> * if working with PHP/HTML/SQL mixed codes to prevent the usage of escape characters

Ok. I guess that doesn't matter for library code like here. For HTML, I can see how it matters, though it seems like bad coding style itself (to me) if HTML and PHP are mixed like that. (OHOT, I also write many applications like that, and then the single quotes are indeed better.) For SQL, I don't really see the advantage of single quotes, I use it with double quotes and even interpolated variables and it never causes unexpected problems.

> * if characters like "\" should be usable, but finally not as an escape character

But in single quotes, backslashes also have to be escaped. There's no difference here.

> * if it's important, because of security reasons, to find accidentally/unwanted added variables within interpolated strings. Especially if the editor does not support the highlighting of variables within a string (like "git diff" results, Atom, Sourcetree, Windows Editor ;-) )

So something like "This costs 12$which is not really cheap." printing the value of `$which`? I guess that's a valid concern.